[Chore] Documentation for ReturningSideEffects (#201)
BendingSpoons/katana-swift: README.md

}
```

> In order to further improve the usability of side effects, there is also a version which can return a value. Note that both the state and dependencies types are erased, to allow for more freedom when using for example in libraries.

Reviewer: ```suggestion
In order to further improve the usability of side effects, there is also a version which can return a value. Note that both the state and dependencies types are erased, to allow for more freedom when using it in libraries, for example.
```

}
```

> Note that the step `0` can be incapsulated in a protocol, should it be a prominent use case for the library/app.

Reviewer: ```suggestion
Note that the step `0` can be encapsulated in a protocol, should it be a prominent use case for the library/app.
```

Reviewer: There is something not convincing in the `should it` form. This is not a question... 🤔 
what about:
 ```suggestion
Note that the step `0` can be encapsulated in a protocol, and that should be a prominent use case for the library/app.
```

Author: I think the `should it be` use is correct, what you proposed has a different meaning form what I intended.

I wanted to say that if using RSE is a prominent use case, then you can use a protocol. (cfr https://context.reverso.net/traduzione/inglese-italiano/Should+it+be)
